forge-gui/res/cardsfolder/upcoming/gandalf_westward_voyager.txt

```diff
@@ -0,0 +1,15 @@
+Name:Gandalf, Westward Voyager
+ManaCost:3 G U
+Types:Legendary Creature Avatar Wizard
+PT:5/5
+T:Mode$ SpellCast | ValidCard$ Card.cmcGE5 | ValidActivatingPlayer$ You | Execute$ TrigReveal | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a spell with mana value 5 or greater, each opponent reveals the top card of their library. If any of those cards shares a card type with that spell, copy that spell, you may choose new targets for the copy, and each opponent draws a card. Otherwise, you draw a card. (A copy of a permanent spell becomes a token.)
+SVar:TrigReveal:DB$ RepeatEach | RepeatPlayers$ Opponent | RepeatSubAbility$ DBReveal | SubAbility$ DBChoose
+SVar:DBReveal:DB$ PeekAndReveal | Defined$ Remembered | ImprintRevealed$ True
```

Comment:
way too complicated + you get to cast your own spell, not the others...

Reply:
you are still copying the wrong thing

Reply:
Oh you copy _your_ spell 🤦

Reply:
the RevealValid still seems wrong

Reply:
you mean the param itself or the content of the param

Reply:
all cards should be revealed
so probably just do it like the card says in the conditions